### PR TITLE
Feature/firebase

### PR DESCRIPTION
**Pull request by MichaelReyesGit**

### CODE REVIEW

File: src/Components/ContractorDetails/ContractorDetails.js [hunk 1]
```diff
@@ -26,11 +27,22 @@ function ContractorDetails() {
   return (
     <div className="contractor-profile-container">
       <div className="container-details">
-        <h3>{contractorProfile.name}</h3>
-        <div className="image">
-          <img className="profile-img" src="https://picsum.photos/200/300" />
+        <div style={{ display: "flex", alignItems: "center" }}>
```
Comment (APinzone-Pursuit) [inline src/Components/ContractorDetails/ContractorDetails.js:30]: Please avoid inline styles.

File: src/Components/Chats/Chats.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
+// import { React, useState, useEffect } from "react";
```
Comment (cdlewis) [inline src/Components/Chats/Chats.js:1]: There's not much reason to commit commented out code.

File: src/Components/Chats/Chats.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        .get("https://api.chatengine.io/users/me/", {
+          headers: {
+            "Project-ID": "4c4dd93c-8b8a-4c2e-8f06-e269442b9589",
+            "User-Name": "michaelreyes@pursuit.org",
+            "User-Secret": "pP1nQFv4NSUWUgrJIrSKQMuK58v2",
+          },
+        })
+        .then(() => setLoading(false))
+        .catch((e) => {
+          //   let formdata = new FormData();
```
Comment (cdlewis) [inline src/Components/Chats/Chats.js:59]: we probably want an error state

File: src/Components/Chats/Chats.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+          //   formdata.append("email", authUser.email);
+          //   formdata.append("username", authUser.email);
+          //   formdata.append("secret", authUser.uid);
+
+          //   axios
+          //     .post("https://api.chatengine.io/users/", formdata, {
+          //       headers: {
+          //         "private-key": "e9414631-c47a-4aea-984b-93403eba4a86",
```
Comment (cdlewis) [inline src/Components/Chats/Chats.js:67]: this is very dangerous from a security standpoint

File: src/Components/Chats/Chats.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+  //   if (!authUser || loading) return <div />;
+
+  return (
+    <div className="chats-page">
+      <ChatEngine
+        height="calc(100vh - 66px)"
+        projectID="4c4dd93c-8b8a-4c2e-8f06-e269442b9589"
+        userName="michaelreyes@pursuit.org"
+        userSecret="pP1nQFv4NSUWUgrJIrSKQMuK58v2"
```
Comment (cdlewis) [inline src/Components/Chats/Chats.js:85]: this is very dangerous from a security standpoint

File: src/Components/Firebase/AuthContext.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+export const AuthProvider = ({ children }) => {
+  const [authUser, setAuthUser] = useState({});
+
+  useEffect(() => {
+    const unsubscribe = onAuthStateChanged(auth, (user) => {
+      setAuthUser(user);
+    });
+
+    return unsubscribe;
+  }, []);
```
Comment (cdlewis) [inline src/Components/Firebase/AuthContext.js:20]: this is a little odd, are we using useEffect to register some kind of change listener?

File: src/Components/Navbar.js [hunk 7]
```diff
@@ -38,6 +42,16 @@ function Nav() {
 
   // console.log(window.scrollY);
 
+  const handleLogout = async () => {
+    try {
+      await signOut(auth);
+      alert(`Logout successful!`);
```
Comment (cdlewis) [inline src/Components/Navbar.js:48]: Good for debugging but not a great user experience :P

File: src/Pages/LogInSignup.jsx [hunk 8]
```diff
@@ -39,34 +44,38 @@ function LoginSignup() {
...
-        if (foundUser) {
-          setUserData(foundUser);
-        } else {
-          console.error("User not found");
-          setError("Invalid credentials. Please try again.");
-          setIsLoggedIn(false);
-        }
+        navigate(`/user-profile/${response.data.id}`);
+        // console.log(userCredentials);
+        // console.log(response.data);
```
Comment (cdlewis) [inline src/Pages/LogInSignup.jsx:58]: this applies to other parts of the diff but generally not worth committing commented out code

**Pull request merged**